cmd/settings.go

```diff
@@ -83,6 +87,83 @@ func (r *Settings) MavenSettings(path string) {
 	}
 }
 
+//
+// ProxySettings set proxy settings.
+func (r *Settings) ProxySettings() (err error) {
+	var http, https string
+	var excluded, noproxy []string
+	http, excluded, err = r.getProxy("http")
+	if err == nil {
+		noproxy = append(
+			noproxy,
+			excluded...)
+	} else {
+		return
+	}
+	https, excluded, err = r.getProxy("https")
+	if err == nil {
+		noproxy = append(
+			noproxy,
+			excluded...)
+	} else {
+		return
+	}
+	for i := range *r {
+		p := &(*r)[i]
+		switch p.Name {
+		case "java":
+			d := p.InitConfig[0].ProviderSpecificConfig
+			if http != "" {
+				d["httpproxy"] = http
+			}
+			if https != "" {
+				d["httpsproxy"] = https
+			}
+			if len(noproxy) > 0 {
+				d["noproxy"] = strings.Join(noproxy, ",")
+			}
+		}
+	}
+	return
+}
+
+//
+// getProxy set proxy settings.
+func (r *Settings) getProxy(kind string) (url string, excluded []string, err error) {
+	var p *api.Proxy
+	var id *api.Identity
+	var user, password string
+	p, err = addon.Proxy.Find(kind)
+	if err != nil {
+		if errors.Is(err, &hub.Conflict{}) {
```

Comment:
Why would the Hub return a conflict on listing the proxies?

Reply:
Meant NotFound instead of Conflict.
Fixed.